doc/nrf/libraries/others/emds.rst

```diff
@@ -0,0 +1,88 @@
+
+The :c:func:`emds_load` will restore the entries into the memory areas from the flash. This has to be done after all the entries has been added.
+
+After restoring previous data the application must run a :c:func:`emds_prepare` to prepare the flash area for receiving new entries. If the remaining empty flash area is smaller than the required data size, the flash area will be automatically erased to increase the available flash area.
+
+The storage will be done in a deterministic time so it will be possible know how long it takes to stored all the registered entries. However this is dependent on the chip, so to measure time is important and :kconfig:option:`CONFIG_EMDS_FLASH_TIME_BASE_OVERHEAD_US`, :kconfig:option:`CONFIG_EMDS_FLASH_TIME_ENTRY_OVERHEAD_US` and :kconfig:option:`CONFIG_EMDS_FLASH_TIME_BASE_OVERHEAD_US` can be configured. It is important when :kconfig:option:`CONFIG_EMDS_FLASH_TIME_BASE_OVERHEAD_US` is changed, the worst time is considered. That including time for erase when doing garbage collection in NVS. If partial erase is not selected/supported it need to include the time of a complete sector erase. When partial erase is selected and supported by HW, it need to include time for the scheduler to trigger and that is dependent on the time of :kconfig:option:`CONFIG_SOC_FLASH_NRF_PARTIAL_ERASE_MS`.
+
+The application must call :c:func:`emds_store` to store all entries provided. This can be only be done once before it need to run :c:func:`emds_prepare` again. When the data storage is complete a callback will be called if provided in :c:func:`emds_init`. This callback notifies the application that the data storage is complete, and can be used to reboot the cpu or execute other function that is needed.
+After completion of :c:func:`emds_store` the :c:func:`emds_is_ready` function call will return false, since it can no longer guarantee that the data will fit in flash area.
+When invoked, :c:func:`emds_store` will trigger the emergency data store process in a separate thread. It will then store all the registered entries.
+Invocation of this call should be performed when the application detects a loss of power, when a reboot is triggered, etc.
+The :c:func:`emds_is_ready` can be called to check if EMDS is prepared to store data. The :c:func:`emds_store` can only be run once, and area has to be prepared before it can be used again.
+
+.. note::
+    Before the application call the :c:func:`emds_store` the application should try to shutdown application specific features that consumes a lot of power.
+    Shutting down these features may prolong the time the cpu is alive, and improve the storage time.
+    For example, if Bluetooth is used, disabling Bluetooth before shutdown will save power and stopping mpsl scheduler will increase storage speed. (See :file:`tests/subsys/emds/emds_api/src/main.c`)
```

Comment:
Why are we referring to this file? Maybe write e.g.:
See the file :file:`tests/subsys/emds/emds_api/src/main.c` for...